### PR TITLE
fix(vault): normalize withdraw limit queue for correctness

### PR DESCRIPTION
**Pull request by 0xferit**
## Summary

The withdraw limit module receives the raw caller-supplied `strategiesParam_` rather than the
resolved queue the vault actually uses. While `_defaultQueue` is governance-trusted (strategies
are placed there by `QUEUE_MANAGER`), the module should still receive accurate data about the
actual execution path for correctness.

This normalizes the queue before passing it to the module in both `_redeem` and `_max_withdraw`.

## Note

The `_defaultQueue` is governance-managed and trusted. This fix is a correctness hardening --
not a security-critical patch. No exploit path exists against current or reasonably expected
module implementations since `_defaultQueue` strategies are governance-approved.

### CODE REVIEW

**Comment by 0xferit** [comment]
@codex review

**Comment by 0xferit** [comment]
@codex review

File: test/unit/core/multistrategyvault/ERC4626.t.sol [hunk 1]
```diff
@@ -14,6 +14,21 @@ import { MockLockedStrategy } from "test/mocks/core/MockLockedStrategy.sol";
...
 import { MockDepositLimitModule } from "test/mocks/core/MockDepositLimitModule.sol";
 
+contract QueueSensitiveWithdrawLimitModule {
+    address public blockedStrategy;
+
+    constructor(address _blockedStrategy) {
+        blockedStrategy = _blockedStrategy;
+    }
+
+    function availableWithdrawLimit(address, uint256, address[] calldata strategies) external view returns (uint256) {
```
[the remark this answers is not in the document]
Comment (0xferit) [reply test/unit/core/multistrategyvault/ERC4626.t.sol:24]: Addressed in latest commit: params renamed to owner/maxLoss for clarity.

**Comment by 0xferit** [comment]
@codex review

File: src/core/MultistrategyVault.sol [hunk 2]
```diff
@@ -2119,11 +2121,6 @@ contract MultistrategyVault is IMultistrategyVault {
             vars.have = vars.currentIdle;
             vars.loss = 0;
 
-            // Determine which strategy queue to use
-            vars.withdrawalStrategies = strategiesParam_.length != 0 && !useDefaultQueue
```
Comment (qGolem) [inline src/core/MultistrategyVault.sol:2123]: I don't understand why this is necessary

**Review by qGolem — CHANGES_REQUESTED** [review]

File: src/core/MultistrategyVault.sol [hunk 3]
```diff
@@ -2106,7 +2109,7 @@ contract MultistrategyVault is IMultistrategyVault {
                     IWithdrawLimitModule(_withdrawLimitModule).availableWithdrawLimit(
```
Comment (qGolem) [inline src/core/MultistrategyVault.sol:2109]: We need a concrete withdrawlimitmodule implementation

**Review by qGolem — CHANGES_REQUESTED** [review]
NACK but we should make withdrawlimitmodule implementation

**Comment by 0xferit** [comment]
Closing -- _defaultQueue is governance-trusted (QUEUE_MANAGER), so the module receiving the raw strategiesParam_ is not a correctness issue. No fix needed.
